src/main/java/com/exactpro/th2/common/schema/factory/CommonFactory.java

```diff
@@ -478,6 +500,120 @@ public static CommonFactory createFromKubernetes(String namespace, String boxNam
         }
     }
 
+    @Override
+    public InputStream loadSingleDictionary() {
+        Path dictionaryFolder = getPathToDictionaryAliasesDir();
+        try {
+            LOGGER.debug("Loading dictionary from folder: {}", dictionaryFolder);
+            List<Path> dictionaries = null;
+            if (Files.isDirectory(dictionaryFolder)) {
+                try (Stream<Path> files = Files.list(dictionaryFolder)) {
+                    dictionaries = files.filter(Files::isRegularFile).collect(Collectors.toList());
+                }
+            }
+
+            if (dictionaries==null || dictionaries.isEmpty()) {
+                throw new IllegalStateException("No dictionary at path: " + dictionaryFolder.toAbsolutePath());
+            } else if (dictionaries.size() > 1) {
+                throw new IllegalStateException("Found several dictionaries at path: " + dictionaryFolder.toAbsolutePath());
+            }
+
+            var targetDictionary = dictionaries.get(0);
+
+            return new ByteArrayInputStream(getGzipBase64StringDecoder().decode(Files.readString(targetDictionary)));
+        } catch (IOException e) {
+            throw new IllegalStateException("Can not read dictionary from path: " + dictionaryFolder.toAbsolutePath(), e);
+        }
+    }
+
+    @Override
+    public Set<String> getDictionaryAliases() {
+        Path dictionaryFolder = getPathToDictionaryAliasesDir();
+        try {
+            if (!Files.isDirectory(dictionaryFolder)) {
+                return Set.of();
+            }
+
+            try (Stream<Path> files = Files.list(dictionaryFolder)) {
+                return files.filter(Files::isRegularFile).map(dictionary -> FilenameUtils.removeExtension(dictionary.getFileName().toString())).collect(Collectors.toSet());
```

Comment:
I think it's more readable:
```
return files
        .filter(Files::isRegularFile)
        .map(dictionary -> FilenameUtils.removeExtension(dictionary.getFileName().toString()))
        .collect(Collectors.toSet());
```
What do you think?
I think it's OK if we have short chain like `files.filter(Files::isRegularFile).collect(Collectors.toList())`.